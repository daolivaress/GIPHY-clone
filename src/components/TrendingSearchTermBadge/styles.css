.trending-search-term-badge {
  display: flex;
  align-items: center;
  height: 50px;
  font-weight: 700;
  text-wrap: nowrap;
  gap: 8px;
  background-image: linear-gradient(0deg, var(--giphy-darkest-grey), var(--giphy-dark-charcoal));
  border-radius: 99px;
  padding: 0 1.5rem;
  cursor: pointer;
}

.trending-search-term-badge img {
  width: 1.5rem;
}
