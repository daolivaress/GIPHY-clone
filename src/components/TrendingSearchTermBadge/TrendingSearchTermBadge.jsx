import "./styles.css";

const TrendingSearchTermBadge = ({ title, onNewCategory }) => {
  return (
    <div className="trending-search-term-badge" onClick={() => {
      onNewCategory(title.trim())
    }}>
      <img src="/src/assets/trending.svg" alt="trending-icon" />
      <p>{title}</p>
    </div>
  );
};

export default TrendingSearchTermBadge;
