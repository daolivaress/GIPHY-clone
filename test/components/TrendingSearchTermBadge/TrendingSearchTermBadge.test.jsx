import { render, screen } from "@testing-library/react";
import TrendingSearchTermBadge from "../../../src/components/TrendingSearchTermBadge/TrendingSearchTermBadge";

describe("pruebas en <TrendingSearchTermBadge/>", () => {
  test("Debe hacer match con el snapshot", () => {
    const { container } = render(
      <TrendingSearchTermBadge title=" " onNewCategory={() => {}} />
    );
    expect(container).toMatchSnapshot();
  });

  test("Debe cambiar de categoria", () => {
    const onNewCategory = jest.fn();
    const title = "hola";
    render(
      <TrendingSearchTermBadge
        title={title}
        onNewCategory={onNewCategory}
      />
    );
    screen.getByTestId("trending-search-term-badge").click();
    expect(onNewCategory).toHaveBeenCalledWith(title);
  });

  test("Debe mostrar el titulo en una etiqueta p", () => {
    render(<TrendingSearchTermBadge title="hola" onNewCategory={() => {}} />);
    expect(screen.getByRole("paragraph").textContent).toBe("hola");
  });

});
