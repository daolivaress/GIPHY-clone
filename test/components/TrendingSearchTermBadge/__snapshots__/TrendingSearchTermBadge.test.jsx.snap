// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <TrendingSearchTermBadge/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="trending-search-term-badge"
    data-testid="trending-search-term-badge"
  >
    <img
      alt="trending-icon"
      src="/src/assets/trending.svg"
    />
    <p>
       
    </p>
  </div>
</div>
`;
